### PR TITLE
fix token detection fallback

### PR DESCRIPTION
**Pull request by bergeron**
## Explanation

Even when account API detection succeeded, it was doing an additional, redundant RPC based detection on the same chains.

## Changelog

- **@metamask/assets-controllers**: Fixes an issue where the token detection was unnecessarily falling back to an RPC approach, causing redundant detections.

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate
- [ ] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

### CODE REVIEW

File: packages/assets-controllers/src/TokenDetectionController.ts [hunk 1]
```diff
@@ -701,26 +701,11 @@ export class TokenDetectionController extends StaticIntervalPollingController<To
...
-      // If API succeeds and no chains are left for RPC detection, we can return early
-      if (
-        apiResult?.result === 'success' &&
-        chainsToDetectUsingRpc.length === 0
-      ) {
-        return;
-      }
-
-      // If API fails or chainsToDetectUsingRpc still has items, add chains to RPC detection
-      this.#addChainsToRpcDetection(
```
Comment (bergeron) [inline packages/assets-controllers/src/TokenDetectionController.ts:719]: This seems like it was just adding all the account API chains, so it was doing an additional RPC based detection even though they just succeeded via account api.


If the account api call failed, we could try to fallback to RPC for all of the chains, although I don't think it's necessary so simplified.

Comment (bergeron) [reply packages/assets-controllers/src/TokenDetectionController.ts:719]: Let me know if either of these fixes makes sense to you @salimtb

Comment (salimtb) [reply packages/assets-controllers/src/TokenDetectionController.ts:719]: yes, i got the idea andthe fix makes sense to me. Just to confirm, given that the unit test is currently failing, with this fix, do we revert to the RPC method if the call to the account API fails ?

Comment (bergeron) [reply packages/assets-controllers/src/TokenDetectionController.ts:719]: I updated it to keep the RPC fallback if the account API request fails.

File: packages/assets-controllers/src/TokenDetectionController.ts [hunk 2]
```diff
@@ -822,10 +816,7 @@ export class TokenDetectionController extends StaticIntervalPollingController<To
         .getMultiNetworksBalances(selectedAddress, chainIds, supportedNetworks)
         .catch(() => null);
 
-      if (
-        !tokenBalancesByChain ||
-        Object.keys(tokenBalancesByChain).length === 0
-      ) {
+      if (tokenBalancesByChain === null) {
```
Comment (bergeron) [inline packages/assets-controllers/src/TokenDetectionController.ts:819]: This used to return `failed` if the api response was empty.  But that's a valid response if it's an empty account with no tokens.

**Review by salimtb — APPROVED** [review]

**Pull request merged**